### PR TITLE
Add MESSAGE_ADD_REACTION to text only permissions

### PR DESCRIPTION
**Pull request by MinnDevelopment**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [x] Internal code
- [ ] Library interface (affecting end-user code) 
- [ ] Documentation
- [ ] Other: \_____ 

Closes Issue: #886 

## Description

You can only use this permissions on text channels

### CODE REVIEW

**Comment by DV8FromTheWorld** [comment]
It feels like the pattern of determining text vs voice based on offset ranges here is becoming an anti-pattern that we have to revisit every time permissions change to validate that it is still correct.

Perhaps we should consider a different way to determining if it is text or voice specific

**Comment by MinnDevelopment** [comment]
@DV8FromTheWorld I agree but I feel like that would be more suited as a version 4 change.

**Comment by DV8FromTheWorld** [comment]
Perhaps, but it wouldn't be a breaking change from the api perspective. It'd probably involve additional constructor params though.

I'm fine with with a v3 or v4 change, but I'd like there to be a change in the codebase.

File: src/main/java/net/dv8tion/jda/core/Permission.java [hunk 1]
```diff
@@ -188,7 +188,7 @@ public boolean isText()
      */
     public boolean isVoice()
     {
-        return offset == 10 || offset == 8 || offset > 19 && offset < 26;
+        return offset == 10 || offset == 8 || (offset > 19 && offset < 26);
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/core/Permission.java]:
```suggestion
        return offset == 8 || offset == 10 || (offset > 19 && offset < 26);
```

**Pull request merged**